components/Cards/src/MDCCard.m

```diff
@@ -221,16 +221,12 @@ - (UIColor *)shadowColorForState:(UIControlState)state {
 }
 
 - (void)setHighlighted:(BOOL)highlighted {
-  [super setHighlighted: highlighted];
-  if (highlighted) {
-    // Note: setHighlighted: can get getting more calls with YES than NO when clicking rapidly.
-    // To guard against ink never going away and darkening our card we call
-    // startTouchEndedAnimationAtPoint:completion:.
-    [self.inkView startTouchEndedAnimationAtPoint:_lastTouch completion:nil];
+  if (highlighted && !self.highlighted) {
     [self.inkView startTouchBeganAnimationAtPoint:_lastTouch completion:nil];
-  } else {
+  } else if (!highlighted && self.highlighted) {
     [self.inkView startTouchEndedAnimationAtPoint:_lastTouch completion:nil];
   }
+  [super setHighlighted:highlighted];
```

Comment:
Did we move this on purpose? can we elaborate why it needs to be after the super call?

Reply:
yes, after super is called, `self.highlighted` is updated to be what `highlighted` is so we lose the previous state.